hummingbot/connector/derivative/okx_perpetual/okx_perpetual_web_utils.py

```diff
@@ -0,0 +1,285 @@
+    api_factory = build_api_factory_without_time_synchronizer_pre_processor(throttler=throttler)
+    rest_assistant = await api_factory.get_rest_assistant()
+    endpoint = CONSTANTS.REST_SERVER_TIME[CONSTANTS.ENDPOINT]
+    url = get_rest_url_for_endpoint(endpoint=endpoint, domain=domain)
+    limit_id = get_rest_api_limit_id_for_endpoint(method=CONSTANTS.REST_SERVER_TIME[CONSTANTS.METHOD],
+                                                  endpoint=endpoint)
+    response = await rest_assistant.execute_request(
+        url=url,
+        throttler_limit_id=limit_id,
+        method=RESTMethod.GET,
+    )
+    server_time = int(response["data"][0]["ts"])
+
+    return server_time
+
+
+def endpoint_from_message(message: Dict[str, Any]) -> Optional[str]:
+    endpoint = None
+    if isinstance(message, dict):
+        event = message.get("event")
+        op = message.get("op")
+        if event is not None:
+            endpoint = event
+        elif op is not None:
+            endpoint = op
+        else:
+            endpoint = message["arg"].get("channel")
+    return endpoint
+
+
+def payload_from_message(message: Dict[str, Any]) -> List[Dict[str, Any]]:
+    return message.get("data", [])
+
+
+def build_api_factory_without_time_synchronizer_pre_processor(throttler: AsyncThrottler) -> WebAssistantsFactory:
+    api_factory = WebAssistantsFactory(throttler=throttler)
+    return api_factory
+
+
+def get_rest_url_for_endpoint(
+        endpoint: str,
+        domain: str = CONSTANTS.DEFAULT_DOMAIN
+):
+    variant = domain if domain else CONSTANTS.DEFAULT_DOMAIN
+    return CONSTANTS.REST_URLS.get(variant) + endpoint
+
+
+def get_rest_api_limit_id_for_endpoint(method: str, endpoint: str) -> str:
+    return f"{method}-{endpoint}"
+
+
+def get_pair_specific_limit_id(method: str, endpoint: str, trading_pair: str) -> str:
+    base_limit_id = get_rest_api_limit_id_for_endpoint(method, endpoint)
+    return f"{base_limit_id}-{trading_pair}"
+
+
+def _wss_url(endpoint: Dict[str, str], connector_variant_label: Optional[str]) -> str:
+    variant = connector_variant_label if connector_variant_label else CONSTANTS.DEFAULT_DOMAIN
+    return endpoint.get(variant)
+
+
+def wss_linear_public_url(connector_variant_label: Optional[str]) -> str:
+    return _wss_url(CONSTANTS.WSS_PUBLIC_URLS, connector_variant_label)
+
+
+def wss_linear_private_url(connector_variant_label: Optional[str]) -> str:
+    return _wss_url(CONSTANTS.WSS_PRIVATE_URLS, connector_variant_label)
+
+
+def build_rate_limits(trading_pairs: Optional[List[str]] = None) -> List[RateLimit]:
+    trading_pairs = trading_pairs or []
+    rate_limits = []
+    domain = CONSTANTS.DEFAULT_DOMAIN
+    rate_limits.extend(_build_websocket_rate_limits(domain))
+    rate_limits.extend(_build_public_rate_limits())
+    rate_limits.extend(_build_private_rate_limits(trading_pairs))
+
+    return rate_limits
+
+
+def _build_websocket_rate_limits(domain: str) -> List[RateLimit]:
```

Comment:
in the next version simplify the construction in one list, as is done in OKX spot. Also move it to constants.